app/controllers/api/hosts_controller.rb

```diff
@@ -11,21 +12,43 @@ class HostsController < BaseController
     include Subcollections::Tags
 
     def edit_resource(type, id, data = {})
+      # TODO: drop 'credentials' parameter field when ui-classic hosts is in react
       credentials = data.delete(CREDENTIALS_ATTR)
+      authentications = data.delete(AUTH_ATTR)
       raise BadRequestError, "Cannot update non-credentials attributes of host resource" if data.any?
       resource_search(id, type).tap do |host|
-        all_credentials = Array.wrap(credentials).each_with_object({}) do |creds, hash|
+        all_credentials = Array.wrap(credentials || []).each_with_object({}) do |creds, hash|
           auth_type = creds.delete(AUTH_TYPE_ATTR) || DEFAULT_AUTH_TYPE
           creds.symbolize_keys!
           creds.reverse_merge!(:userid => host.authentication_userid(auth_type))
           hash[auth_type.to_sym] = creds
         end
+        all_credentials, _ = authentications_params_to_authentications(authentications) if authentications
```

Comment:
Is this overwriting `all_credentials` from up on line 20?  Hard to follow how authentications / credentials are used.

Reply:
thanks. I added comments.

The old way passed credentials.
the new way passes authentications

Yes, if they follow the new way, then we are ignoring the old way.
In theory, they will not pass the old way and the new way

Reply:
once we swap out ui classic, we can drop the old way